cloudinit/handlers/jinja_template.py

```diff
@@ -117,15 +116,14 @@ def render_jinja_payload_from_file(
         )
     try:
         instance_data = load_json(load_text_file(instance_data_file))
-    except Exception as e:
-        msg = "Loading Jinja instance data failed"
-        if isinstance(e, (IOError, OSError)):
-            if e.errno == EACCES:
-                msg = (
-                    "Cannot render jinja template vars. No read permission on"
-                    " '%s'. Try sudo" % instance_data_file
-                )
+    except PermissionError as e:
+        msg = (
+            "Cannot render jinja template vars. No read permission on"
+            " '%s'. Try sudo" % instance_data_file
+        )
         raise JinjaLoadError(msg) from e
+    except (OSError, ValueError) as e:
```

Comment:
Oops, I forgot TypeError can be raised from `load_json()`. I think that this should cover the possible exceptions thrown here, but I'll add a top level handler just in case.